client/state/sharing/publicize/actions.js

```diff
@@ -113,12 +115,44 @@ export function updateSiteConnection( connection, attributes ) {
 }
 
 /**
+ * Triggers a network request to refresh a Publicize connection for the
+ * specifiedsite ID.
+ *
+ * @param  {Number} siteId           Site ID
+ * @param  {Object} connection       Connection to be deleted.
+ * @param  {String} connection.label Name of the service that was connected.
+ * @return {Function}                Action thunk
+ */
+export function refreshSiteConnection( siteId, connection ) {
+	return ( dispatch ) => {
+		dispatch( {
+			type: PUBLICIZE_CONNECTIONS_REQUEST,
+			siteId,
+		} );
+
+		return wpcom.undocumented().siteConnections( siteId )
+			.then( ( response ) => {
+				dispatch( receiveConnections( siteId, response ) );
+				dispatch( {
+					type: PUBLICIZE_CONNECTION_REFRESH,
```

Comment:
Oh I didn't notice the singular, since I was looking at the wpcom usage. 👍 

> There is an endpoint to request a single connection but there is no wpcom.undocument wrapper for it. Would creating one be preferable?

This could make sense if the response is heavy. But optional if you think it doesn't make sense.

I would suggest having actions with:
```
fetchConnections( siteId );
fetchConnection( siteId, connectionId );
```

with action types:
```
PUBLICIZE_CONNECTION_REQUEST
PUBLICIZE_CONNECTION_RECEIVE
PUBLICIZE_CONNECTION_REQUEST_SUCCESS
PUBLICIZE_CONNECTION_REQUEST_FAILURE
```

> Should the action names for all four (create/refresh/update/delete) operations be *_SUCCESS and *_FAILURE? I couldn't really find a consistant naming convention there

In this case `refresh` sounds really similar to fetch or request, so I was surprised to see it being dispatched after the request was made.